### PR TITLE
fixed feature issue-2: id-plugin

### PR DESCRIPTION
**Pull request by dianavile**
Hi Nico,

Please merge my branch (issue-2: id-plugin) to your master branch.

kind regards,
Diana Vilé

### CODE REVIEW

File: plugins/id/test.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      |-- -- -- -- -- -- -- -- --|-- -- -- -- -- -- -- -- --|
+      | Table                    | Table <?: #td-id ?>      |
+    `
+    const doc = new Document(input)
+    doc.config.use(plugin)
+    const html = doc.render()
+    expect(html)
+      .to.have.selector('p#td-id')
+  })
+})
```
[the remark this answers is not in the document]
Comment (dianavile) [reply plugins/id/test.js]: Nico,

I added the required Newline, as suggested by houndci-bot, but it does not seem to work.

[the remark this answers is not in the document]
Comment (benevbright) [reply plugins/id/test.js]: @hagenburger

File: plugins/id/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# PIMD ID Plugin
+
+Add ID to HTML classes to code blocks and other elements.
```
Comment (hagenburger) [inline plugins/id/README.md]: “Adds an ID to code blocks and other elements.”

File: plugins/id/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+Lorem ipsum dolor sit amet. <?: #my-paragraph ?>
+````
+
+Results in:
+
+``` html
+<h1 id="my-headline">Headline</h1>
+
+<div id="pimd-example my-code">
```
Comment (hagenburger) [inline plugins/id/README.md]: This is wrong, please check again.

File: plugins/id/package.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "@pimd/classes-plugin",
+  "version": "0.1.0",
+  "description": "Add ID to HTML elements and codeblocks",
```
Comment (hagenburger) [inline plugins/id/package.json]: “Adds an ID …“

File: plugins/id/package.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  "main": "index.js",
+  "author": "Nico Hagenburger <nico@hagenburger.net> (https://twitter.com/hagenburger)",
+  "keywords": [
+    "markdown",
+    "pimd-plugin"
+  ],
```
Comment (hagenburger) [inline plugins/id/package.json]: Contributors are missing.

File: plugins/id/test.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
+const {Document} = require('../..')
+const plugin = require('.')
+
+describe('ID-plugin', () => {
```
Comment (hagenburger) [inline plugins/id/test.js]: No dash for “ID plugin”.

File: plugins/id/test.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  it('should add an id to a headline', () => {
```
Comment (hagenburger) [inline plugins/id/test.js]: “ID” in all caps.

File: plugins/id/test.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+      # Test <?: #my-id ?>
+    `
+    const doc = new Document(input)
+    doc.config.use(plugin)
+    const html = doc.render()
+    expect(html)
+      .to.have.selector('h1#my-id')
+  })
+
+  it('should add an id to a paragraph', () => {
```
Comment (hagenburger) [inline plugins/id/test.js]: “ID” in all caps.

File: plugins/id/test.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+      p Test <?: #my-id ?>
+    `
+    const doc = new Document(input)
+    doc.config.use(plugin)
+    const html = doc.render()
+    expect(html)
+      .to.have.selector('p#my-id')
+  })
+
+  it('should add an id to a code block', () => {
```
Comment (hagenburger) [inline plugins/id/test.js]: “ID” in all caps.

File: plugins/id/test.js [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    const input = unindent`
+      ~~~ html #my-id
+      <p> Lorem ipsum dolor sit amet.</p>
```
Comment (hagenburger) [inline plugins/id/test.js]: Remove space after `<p>`

File: plugins/id/test.js [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+      ~~~
+    `
+    const doc = new Document(input)
+    doc.config.use(plugin)
+    const html = doc.render()
+    expect(html)
+      .to.have.selector('div#my-id')
+  })
+
+  it('should add an id to a table', () => {
```
Comment (hagenburger) [inline plugins/id/test.js]: “ID” in all caps.

File: plugins/id/test.js [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+      | Table                    | Table <?: #td-id ?>      |
+    `
+    const doc = new Document(input)
+    doc.config.use(plugin)
+    const html = doc.render()
+    expect(html)
+      .to.have.selector('p#td-id')
+  })
+})
+
```
[the remark this answers is not in the document]
Comment (hagenburger) [reply plugins/id/test.js]: @benevbright can you help them with the newline issue? I think their editor needs to be configured.

[the remark this answers is not in the document]
Comment (benevbright) [reply plugins/id/test.js]: Oh, OK @hagenburger I will! This hound dog is using same `.eslintrc` file, right?

[the remark this answers is not in the document]
Comment (hagenburger) [reply plugins/id/test.js]: @benevbright yes: https://github.com/hagenburger/pimd/blob/master/.hound.yml

You can run `npm run lint` from your command line.

**Comment by hagenburger** [comment]
Closing this as duplicate of #13.

**Pull request closed without merging**